### PR TITLE
fix(TextField): avoid right helper text from wrapping, fix spacing and aria label for maxCount text when multiline is true

### PR DESCRIPTION
**Pull request by marcoskolodny**
Issues:
- [WEB-2009](https://jira.tid.es/browse/WEB-2009)
- [WEB-2070](https://jira.tid.es/browse/WEB-2070)

### CODE REVIEW

File: src/text-field-components.css.ts [hunk 1]
```diff
@@ -126,6 +126,8 @@ export const leftHelperText = style([
     },
 ]);
 
+export const rightHelperText = sprinkles({paddingLeft: 16});
```
Comment (marcoskolodny) [inline src/text-field-components.css.ts:129]: Avoid this issue:

Screenshot from 2024-10-18 15-27-18

File: src/text-field-components.tsx [hunk 2]
```diff
@@ -106,8 +106,8 @@ export const HelperText = ({leftText, rightText, error, id}: HelperTextProps): J
                 </p>
             )}
             {rightText && (
-                <div className={classnames(styles.helperText)}>
-                    <Text1 color={rightColor} regular as="p" textAlign="right">
+                <div className={classnames(styles.helperText, {[styles.rightHelperText]: !!leftText})}>
+                    <Text1 color={rightColor} regular as="p" textAlign="right" wordBreak={false}>
```
Comment (marcoskolodny) [inline src/text-field-components.tsx:110]: Before/after:

input-fields-screenshot-test-tsx-text-field-multiline-and-max-length-with-long-helper-text-1-diff

File: src/text-field-components.tsx [hunk 3]
```diff
@@ -106,8 +114,20 @@ export const HelperText = ({leftText, rightText, error, id}: HelperTextProps): J
...
+                        <ScreenReaderOnly>
+                            <span>{rightTextLabel}</span>
+                        </ScreenReaderOnly>
+                    )}
+                    <Text1
+                        color={rightColor}
+                        regular
+                        as="p"
+                        textAlign="right"
+                        wordBreak={false}
```
Comment (marcoskolodny) [inline src/text-field-components.tsx:128]: Avoid wrapping

**Review by atabel — APPROVED** [review]

**Review by pladaria — APPROVED** [review]

**Pull request merged**